src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -1,140 +1,183 @@
+        self.response_channel = f"{self.channel_prefix}/response"
+        self.events_channel = f"{self.response_channel}/events/"
         self._setup_redis_connection()
         self.area_uuid_markets_mapping = {}
-        self.markets_mapping = {}
+        self.markets_mapping = {}  # Dict[market_id: market] mapping
         self.recommendations = []
 
     def _setup_redis_connection(self):
         self.myco_ext_conn = ResettableCommunicator()
         self.myco_ext_conn.sub_to_multiple_channels(
-            {"external-myco/get-simulation-id": self.get_simulation_id,
+            {"external-myco/get-simulation-id": self.publish_simulation_id,
              f"{self.channel_prefix}offers-bids/": self.publish_offers_bids,
              f"{self.channel_prefix}post-recommendations/": self.match_recommendations})
 
     def publish_offers_bids(self, message):
         """Publish open offers and bids.
 
-        published data are of the following format
-        market_offers_bids_list_mapping = {"market_id" : {"bids": [], "offers": [] }, }
+        published data are of the following format:
+            {"bids_offers": {`market_id` : {"bids": [], "offers": [] }, }}
         """
-        # TODO: message can contain filters
-        data = {"event": "offers_bids_response"}
+        response_data = {"event": "offers_bids_response"}
+        data = json.loads(message.get("data"))
+        filters = data.get("filters", {})
+        # IDs of markets (Areas) the client is interested in
+        filtered_area_ids = filters.get("markets", None)
         market_offers_bids_list_mapping = {}
-        for area_uuid, markets in self.area_uuid_markets_mapping.items():
-            for market in markets:
-                self.markets_mapping[market.id] = market
-                market_offers_bids_list_mapping[market.id] = {"bids": [], "offers": []}
-                bids, offers = market.open_bids_and_offers
-                market_offers_bids_list_mapping[market.id]["bids"].extend(
-                    list(bid.serializable_dict() for bid in bids.values()))
-                market_offers_bids_list_mapping[market.id]["offers"].extend(
-                    list(offer.serializable_dict() for offer in offers.values()))
-        data.update({
-            "market_offers_bids_list_mapping": market_offers_bids_list_mapping,
+        for area_id, market_slots in self.area_uuid_markets_mapping.items():
+            if filtered_area_ids and area_id not in filtered_area_ids:
+                # Client is uninterested in this Area -> skip
+                continue
+            for market_slot in market_slots:
+                # Cache the market (needed while matching)
+                self.markets_mapping[market_slot.id] = market_slot
+                bids_list, offers_list = self._get_bids_offers(market_slot, filters)
+                market_offers_bids_list_mapping[market_slot.id] = {
+                    "bids": bids_list, "offers": offers_list}
```

Comment:
I vote to rename `market_slots` into simple `markets` and `market_slot` into `market`.
It is kind of a best practice in the d3a code. `market_slot` usually only refers to the time of the market. Thanks!